lib/auth/grpcserver_test.go

```diff
@@ -1653,3 +1653,21 @@ func TestChangeUserAuthenticationRateLimiting(t *testing.T) {
 	_, err = clt.ChangeUserAuthentication(ctx, &proto.ChangeUserAuthenticationRequest{})
 	require.True(t, trace.IsLimitExceeded(err))
 }
+
+func TestStartAccountRecoveryRateLimiting(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	srv := newTestTLSServer(t)
+
+	clt, err := srv.NewClient(TestNop())
+	require.NoError(t, err)
+
+	// Max rate limit.
+	for i := 0; i < 10; i++ {
+		_, err = clt.StartAccountRecovery(ctx, &proto.StartAccountRecoveryRequest{})
+		require.Error(t, err)
+	}
+
+	_, err = clt.StartAccountRecovery(ctx, &proto.StartAccountRecoveryRequest{})
+	require.True(t, trace.IsLimitExceeded(err))
```

Comment:
Suggestion: instead of hard-coding the limit in the test, loop until we hit _a_ limit? Makes the test more resilient to changes, while still making sure a rate limit exists.

```suggestion
	var gotErr error
	const maxAttempts = 100
	for i := 0; i < maxAttempts; i++ {
		_, err = clt.StartAccountRecovery(ctx, &proto.StartAccountRecoveryRequest{})
		if err != nil {
			gotErr = err
			break
		}
	}
	require.True(t, trace.IsLimitExceeded(gotErr))
```

Reply:
I stayed with the lower attempt but looped until I expected to get a limit exceeded